protos/backend_service.proto

```diff
@@ -0,0 +1,127 @@
+    rpc GetWorkItems(stream GetWorkItemsRequest) returns (stream WorkItem);
+
+    // Gets orchestration runtime state (history, etc.) for a given orchestration instance.
+    rpc GetOrchestrationRuntimeState (GetOrchestrationRuntimeStateRequest) returns (GetOrchestrationRuntimeStateResponse);
+
+    // Completes an outstanding activity work item and adds a new event to the target orchestration's inbox.
+    rpc CompleteActivityWorkItem (CompleteActivityWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Abandons an outstanding activity work item. Abandoned work items will be delivered again after some delay.
+    rpc AbandonActivityWorkItem (AbandonActivityWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Completes an outstanding orchestrator work item, and adds a new event to the target orchestration's inbox.
+    rpc CompleteOrchestrationWorkItem (CompleteOrchestrationWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Abandons an outstanding orchestrator work item. Abandoned work items will be delivered again after some delay.
+    rpc AbandonOrchestrationWorkItem (AbandonOrchestrationWorkItemRequest) returns (google.protobuf.Empty);
+
+    // Fetches metadata about the connected service. This may also be used to verify connectivity.
+    rpc Metadata(google.protobuf.Empty) returns (google.protobuf.Empty); // TODO: Payloads
+}
+
+// Request payload for adding new orchestration events.
+message AddOrchestrationEventRequest {
+    // The ID of the orchestration to send an event to.
+    OrchestrationInstance instance = 1;
+    // The event to send to the orchestration.
+    HistoryEvent event = 2;
+}
+
+// Request parameters for fetching orchestration runtime state.
+message GetOrchestrationRuntimeStateRequest {
+    // The ID of the target orchestration instance.
+    OrchestrationInstance instance = 1;
+}
+
+// Response payload returned when fetching orchestration runtime state.
+message GetOrchestrationRuntimeStateResponse {
+    // The existing history events for the target orchestration instance.
+    repeated HistoryEvent history = 1;
+}
+
+// Request payload for completing an activity work item.
+message CompleteActivityWorkItemRequest {
+    // The completion token that was provided when the work item was fetched.
+    string completionToken = 1;
+
+    // The response event that will be sent to the orchestrator.
+    // This must be either a TaskCompleted event or a TaskFailed event.
+    HistoryEvent responseEvent = 2;
+}
+
+// Request payload for abandoning an activity work item.
+message AbandonActivityWorkItemRequest {
+    // The completion token that was provided when the work item was fetched.
+    string completionToken = 1;
+}
+
+// Request payload for completing an orchestration work item.
+message CompleteOrchestrationWorkItemRequest {
+    // The completion token that was provided when the work item was fetched.
+    string completionToken = 1;
```

Comment:
Note protobuf convention is to use snake_case for all fields

Reply:
Yes, but orchestration_service.proto, which this depends on, unfortunately already broke this convention. We can either make everything consistent (which is what the current PR does) or we can update just this file with the general convention and have two different conventions in the same code base. My default preference is to have consistency across the repo.